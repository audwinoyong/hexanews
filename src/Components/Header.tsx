import * as React from "react";
import { Link } from "react-router-dom";

import { withStyles } from "@material-ui/core/styles";

import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";

import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";

const styles = {
  flex: {
    flexGrow: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  },
  root: {
    flexGrow: 1
  }
};

interface IProps {
  styles: any;
}

class Header extends React.Component<IProps> {
  public render() {
    const componentStyles = this.props.styles;

    return (
      <div className={componentStyles.root}>
        <AppBar position="static">
          <Toolbar>
            <IconButton
              className={componentStyles.menuButton}
              color="inherit"
              aria-label="Menu"
            >
              <MenuIcon />
            </IconButton>
            <Typography
              variant="title"
              color="inherit"
              className={componentStyles.flex}
            >
              News
              <ul>
                <li>
                  <Link to="/">Hexanews</Link>
                </li>
                <li>
                  <Link to="/profile">Profile</Link>
                </li>
              </ul>
            </Typography>
            <Button color="inherit">Login</Button>
          </Toolbar>
        </AppBar>
      </div>
    )
  }
}

export default withStyles(styles)(Header as any);
