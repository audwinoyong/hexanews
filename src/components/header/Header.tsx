/**
 * The Header Component that will be displayed in the top of application
 */

import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { Grid } from '@material-ui/core'
import { Link } from 'react-router-dom'
import { push } from 'connected-react-router'
import { translate } from 'react-i18next'
import * as React from 'react'
import AppBar from '@material-ui/core/AppBar'
import CreateIcon from '@material-ui/icons/Create'
import ExitIcon from '@material-ui/icons/ExitToApp'
import IconButton from '@material-ui/core/IconButton'
import MenuItem from '@material-ui/core/MenuItem'
import SearchIcon from '@material-ui/icons/Search'
import Select from '@material-ui/core/Select'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'

import { HEXANEWS_TITLE } from '../../constants/config'
import * as actions from './actions'
import * as Config from '../../constants/config'
import AuthenticationDialog from '../authenticationDialog/AuthenticationDialog'
import i18n from '../../i18n'
import selector, { StateProps } from './selector'
import styles from './styles'
import SuccessSnackbar from '../successSnackbar/SuccessSnackbar'

/**
 * All props required by the components
 */
export type Props = {
  dispatch: Dispatch<any>,
  isLoadingServer: boolean,
} & StateProps

/**
 * All state required by the components
 */
export interface ComponentState {
  isForgotPasswordSnackbarOpen: boolean
  isLoginSnackbarOpen: boolean
  isLogoutSnackbarOpen: boolean
  isRegisterSnackbarOpen: boolean
  language: string
  showDialog: boolean
}

export class Header extends React.Component<Props, ComponentState> {

  /**
   * Main constructor for Header and initialize state
   * @param props props required in the components
   */
  constructor(props: Props) {
    super(props)
    const localLanguage = localStorage.getItem('language')
    const currentLanguage = localLanguage !== null ? localLanguage : 'en'
    this.state = {
      isForgotPasswordSnackbarOpen: false,
      isLoginSnackbarOpen: false,
      isLogoutSnackbarOpen: false,
      isRegisterSnackbarOpen: false,
      language: currentLanguage,
      showDialog: false,
    }
  }

  /**
   * Function that handle onChangeSelection language and put into localStorage
   * @param event contain the language value from textField
   */
  handleLanguageChange = (event: any) => {
    this.setState({ language: event.target.value }, () => {
      i18n.changeLanguage(event.target.value)
      localStorage.setItem('language', event.target.value)
    })
  }

  /**
   * Function that will dispatch an action to logout the current user1
   */
  onLogoutClick = async () => {
    const { dispatch } = this.props
    await dispatch(actions.logout())

    this.setState({ isLogoutSnackbarOpen: true })
  }

  /**
   * Function to show authentication dialog
   */
  handleShowDialog = () => {
    this.setState({ showDialog: true })
  }

  /**
   * Function to close authentication dialog
   */
  handleCloseDialog = () => {
    this.setState({ showDialog: false })
  }

  /**
   * Function to close logout snackbar
   */
  handleCloseLogoutSnackbar = () => {
    this.setState({ isLogoutSnackbarOpen: false })
  }

  /**
   * Function to open login snackbar
   */
  handleOpenLoginSnackbar = () => {
    this.setState({ isLoginSnackbarOpen: true })
  }

  /**
   * Function to close login snackbar
   */
  handleCloseLoginSnackbar = () => {
    this.setState({ isLoginSnackbarOpen: false })
  }

  /**
   * Function to open register snackbar
   */
  handleOpenRegisterSnackbar = () => {
    this.setState({ isRegisterSnackbarOpen: true })
  }

  /**
   * Function to close register snackbar
   */
  handleCloseRegisterSnackbar = () => {
    this.setState({ isRegisterSnackbarOpen: false })
  }

  /**
   * Function to open forgot password snackbar
   */
  handleOpenForgotPasswordSnackbar = () => {
    this.setState({ isForgotPasswordSnackbarOpen: true })
  }

  /**
   * Function to close forgot password snackbar
   */
  handleCloseForgotPasswordSnackbar = () => {
    this.setState({ isForgotPasswordSnackbarOpen: false })
  }

  /**
   * Function to render all available header snackbar
   */
  renderAllHeaderSnackbar = () => {
    const {
      isForgotPasswordSnackbarOpen,
      isLoginSnackbarOpen,
      isLogoutSnackbarOpen,
      isRegisterSnackbarOpen,
    } = this.state
    return [
      <SuccessSnackbar
        key='logoutSnackbar'
        isSnackbarOpen={isLogoutSnackbarOpen}
        message={i18n.t('header.logoutSuccess')}
        handleClose={this.handleCloseLogoutSnackbar}
      />,
      <SuccessSnackbar
        key='loginSnackbar'
        isSnackbarOpen={isLoginSnackbarOpen}
        message={i18n.t('loginForm.loginSuccess')}
        handleClose={this.handleCloseLoginSnackbar}
      />,
      <SuccessSnackbar
        key='registerSnackbar'
        isSnackbarOpen={isRegisterSnackbarOpen}
        message={i18n.t('registerForm.registerSuccess')}
        handleClose={this.handleCloseRegisterSnackbar}
      />,
      <SuccessSnackbar
        key='forgotPasswordSnackbar'
        isSnackbarOpen={isForgotPasswordSnackbarOpen}
        message={i18n.t('forgotPasswordForm.forgotPasswordSuccess')}
        handleClose={this.handleCloseForgotPasswordSnackbar}
      />,
    ]
  }

  /**
   * Function to render right container header
   */
  renderRightContainer = () => {
    const { language } = this.state
    const { isLoggedIn, isLoadingServer, dispatch } = this.props
    return (
      <Grid style={styles.rightContainer as any}>
        {!isLoadingServer &&
          <Grid style={styles.innerRightContainer as any}>
            <Select
              value={language}
              onChange={this.handleLanguageChange}
              name='language'
              displayEmpty
              style={styles.languageSelection}
              disableUnderline
            >
              <MenuItem value='en'>EN</MenuItem>
              <MenuItem value='id'>ID</MenuItem>
            </Select>
            {!isLoggedIn &&
              <Typography
                variant='title'
                color='inherit'
                style={styles.profileTypography}
              >
                <span style={styles.signLink} onClick={this.handleShowDialog}>{i18n.t('header.signIn')}</span>
              </Typography>
            }
            {isLoggedIn &&
              <Typography
                variant='title'
                color='inherit'
                style={styles.profileTypography}
              >
                <Link to={Config.HEADER_LINK.profile} style={styles.profileLink}>{i18n.t('header.profile')}</Link>
              </Typography>
            }
            <IconButton
              style={styles.menuButton}
              color='default'
              aria-label='Search'
              onClick={() => dispatch(push(Config.HEADER_LINK.searchArticle))}
            >
              <SearchIcon style={styles.blackIcon} />
            </IconButton>
            {isLoggedIn &&
              <IconButton
                style={styles.menuButton}
                color='default'
                aria-label='Create'
                onClick={() => dispatch(push(Config.HEADER_LINK.createArticle))}
              >
                <CreateIcon style={styles.blackIcon} />
              </IconButton>
            }
            {isLoggedIn &&
              <IconButton
                style={styles.menuButton}
                color='default'
                aria-label='Logout'
                onClick={this.onLogoutClick}
              >
                <ExitIcon style={styles.blackIcon} />
              </IconButton>
            }
          </Grid>
        }
      </Grid>
    )
  }

  /**
   * Render Header component
   */
  render() {
    const { showDialog } = this.state
    return (
      <Grid style={styles.root}>
        <AppBar position='static' style={styles.appBar}>
          <Toolbar>
            <Grid style={styles.leftContainer} />
            <Grid style={styles.titleContainer}>
              <Typography
                variant='display1'
                color='inherit'
                style={styles.titleTypography as any}
              >
                <Link to='/' style={styles.titleLink}>
                  {HEXANEWS_TITLE}
                </Link>
              </Typography>
            </Grid>
            {this.renderRightContainer()}
          </Toolbar>
        </AppBar>
        <AuthenticationDialog
          handleCloseDialog={this.handleCloseDialog}
          handleOpenForgotPasswordSnackbar={this.handleOpenForgotPasswordSnackbar}
          handleOpenLoginSnackbar={this.handleOpenLoginSnackbar}
          handleOpenRegisterSnackbar={this.handleOpenRegisterSnackbar}
          showDialog={showDialog}
        />
        {this.renderAllHeaderSnackbar()}
      </Grid>
    )
  }
}

const ConnectedHeader = connect(selector)(Header)

export default translate('translations')(ConnectedHeader)