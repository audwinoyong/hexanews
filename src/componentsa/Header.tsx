import * as React from 'react'
import { Link } from 'react-router-dom'

import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import IconButton from '@material-ui/core/IconButton'
import MenuIcon from '@material-ui/icons/Menu'

type Props = {

}

const styles = {
  flex: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: 'white',
    boxShadow: 'none',
    borderBottomWidth: 1,
    borderStyle: 'solid',
    borderBottomColor: '#DDD',
  },
  titleLink: {
    textDecoration: 'none',
    color: '#333',
    marginLeft: 24,
  },
  profileLink: {
    textDecoration: 'none',
    color: '#333',
    marginRight: 64,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  root: {
    flexGrow: 1,
  },
}

class Header extends React.Component<Props> {

  constructor(props: Props) {
    super(props)
  }

  public render() {
    return (
      <div style={styles.root}>
        <AppBar position='static' style={styles.appBar}>
          <Toolbar>
            <IconButton
              style={styles.menuButton}
              color='inherit'
              aria-label='Menu'
            >
              <MenuIcon />
            </IconButton>
            <Typography
              variant='title'
              color='inherit'
              style={styles.flex}
            >
              <Link to='/' style={styles.titleLink}>Hexanews</Link>
            </Typography>
            <Typography
              variant='title'
              color='inherit'
            >
              <Link to='/profile' style={styles.profileLink}>Profile</Link>
            </Typography>
          </Toolbar>
        </AppBar>
      </div>
    )
  }
}

export default Header