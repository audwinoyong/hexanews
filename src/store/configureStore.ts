/**
 * Configuration file for redux store
 */

import thunk from 'redux-thunk'
import { applyMiddleware, createStore } from 'redux'
import { createLogger } from 'redux-logger'
import { connectRouter, routerMiddleware } from 'connected-react-router'
import { composeWithDevTools } from 'remote-redux-devtools'
import { History } from 'history'

import rootReducer from '../reducers'

const isDebuggingInChrome = !!window.navigator.userAgent

/**
 * Function to create the initial store state for application
 * @param initialState the initial state for the store
 * @param history the navigation variable for the application
 */
export default function (initialState = {}, history: History) {
  return createStore(
    connectRouter(history)(rootReducer),
    initialState,
    composeWithDevTools(applyMiddleware(
      routerMiddleware(history),
      thunk,
      createLogger({
        predicate: () => isDebuggingInChrome,
        collapsed: true,
        duration: true,
      }),
    )),
  )
}