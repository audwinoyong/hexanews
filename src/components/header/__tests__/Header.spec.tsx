/**
 * Unit tests for checking the behavior of header components
 */

import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import rootReducer from '../../../reducers'
import { Header, Props, ComponentState } from '../Header'

const mockStore = configureStore([thunk])

Enzyme.configure({ adapter: new Adapter() })

describe('Header', () => {
  it('renders Header', () => {
    const store = mockStore(rootReducer(undefined as any, { type: '' }))
    const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Header
            dispatch={jest.fn()}
            isLoadingServer={false}
            isLoggedIn={false}
          />
        </Provider>
      </BrowserRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders Header with isLoggedIn true', () => {
    const store = mockStore(rootReducer(undefined as any, { type: '' }))
    const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Header
            dispatch={jest.fn()}
            isLoadingServer={false}
            isLoggedIn={false}
          />
        </Provider>
      </BrowserRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handle handleShowDialog and handleCloseDialog Change', () => {
    const dispatch = jest.fn()
    const component = (
      <Header
        dispatch={dispatch}
        isLoadingServer={false}
        isLoggedIn={false}
      />
    )

    const layout = shallow<Props, ComponentState>(component)
    const instance: Header = layout.instance() as Header

    const showDialog = layout.state().showDialog
    expect(showDialog).toBe(false)

    instance.handleShowDialog()
    layout.update()

    const updatedShowDialog = layout.state().showDialog
    expect(updatedShowDialog).toBe(true)

    instance.handleCloseDialog()
    layout.update()

    const updatedShowDialog2 = layout.state().showDialog
    expect(updatedShowDialog2).toBe(false)
  })

  it('handle logout action correctly', () => {
    const dispatch = jest.fn()
    const component = (
      <Header
        dispatch={dispatch}
        isLoadingServer={false}
        isLoggedIn={false}
      />
    )

    const layout = shallow(component)
    const instance: Header = layout.instance() as Header

    instance.onLogoutClick()
    layout.update()

    expect(dispatch).toHaveBeenCalled()
  })
})