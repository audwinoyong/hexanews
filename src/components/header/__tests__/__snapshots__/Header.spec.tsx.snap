// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header 1`] = `
<div
  className=""
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <header
    className="MuiPaper-root-106 MuiPaper-elevation4-112 MuiAppBar-root-98 MuiAppBar-positionStatic-102 MuiAppBar-colorPrimary-104"
    style={
      Object {
        "backgroundColor": "white",
        "boxShadow": "none",
        "marginTop": "1%",
      }
    }
  >
    <div
      className="MuiToolbar-root-133 MuiToolbar-regular-135 MuiToolbar-gutters-134"
    >
      <div
        className=""
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <div
        className=""
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <h1
          className="MuiTypography-root-137 MuiTypography-display1-141 MuiTypography-colorInherit-156"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <a
            href="/"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "fontWeight": 600,
                "textDecoration": "none",
              }
            }
          >
            Hexanews
          </a>
        </h1>
      </div>
      <div
        className=""
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <div
          className=""
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <div
            className="MuiInput-root-167"
            style={
              Object {
                "marginRight": 16,
                "marginTop": 8,
              }
            }
          >
            <div
              className="MuiSelect-root-162"
            >
              <div
                aria-haspopup="true"
                aria-owns={null}
                aria-pressed="false"
                className="MuiSelect-select-163 MuiSelect-selectMenu-164 MuiInput-input-175"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                EN
              </div>
              <input
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id={undefined}
                name="language"
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value="en"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-180 MuiSelect-icon-166"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <h2
            className="MuiTypography-root-137 MuiTypography-title-143 MuiTypography-colorInherit-156"
            style={
              Object {
                "alignSelf": "center",
                "marginLeft": 8,
              }
            }
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#333",
                  "cursor": "pointer",
                  "marginRight": 24,
                  "textDecoration": "none",
                }
              }
            >
              Sign In
            </span>
          </h2>
          <button
            aria-label="Search"
            className="MuiButtonBase-root-197 MuiIconButton-root-191"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "marginLeft": -12,
                "marginRight": 20,
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-196"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-180"
                color={undefined}
                focusable="false"
                style={
                  Object {
                    "color": "rgb(51, 51, 51)",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-200"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header renders Header with isLoggedIn true 1`] = `
<div
  className=""
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <header
    className="MuiPaper-root-106 MuiPaper-elevation4-112 MuiAppBar-root-98 MuiAppBar-positionStatic-102 MuiAppBar-colorPrimary-104"
    style={
      Object {
        "backgroundColor": "white",
        "boxShadow": "none",
        "marginTop": "1%",
      }
    }
  >
    <div
      className="MuiToolbar-root-133 MuiToolbar-regular-135 MuiToolbar-gutters-134"
    >
      <div
        className=""
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <div
        className=""
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <h1
          className="MuiTypography-root-137 MuiTypography-display1-141 MuiTypography-colorInherit-156"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <a
            href="/"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "fontWeight": 600,
                "textDecoration": "none",
              }
            }
          >
            Hexanews
          </a>
        </h1>
      </div>
      <div
        className=""
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <div
          className=""
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <div
            className="MuiInput-root-167"
            style={
              Object {
                "marginRight": 16,
                "marginTop": 8,
              }
            }
          >
            <div
              className="MuiSelect-root-162"
            >
              <div
                aria-haspopup="true"
                aria-owns={null}
                aria-pressed="false"
                className="MuiSelect-select-163 MuiSelect-selectMenu-164 MuiInput-input-175"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                EN
              </div>
              <input
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id={undefined}
                name="language"
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value="en"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-180 MuiSelect-icon-166"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <h2
            className="MuiTypography-root-137 MuiTypography-title-143 MuiTypography-colorInherit-156"
            style={
              Object {
                "alignSelf": "center",
                "marginLeft": 8,
              }
            }
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#333",
                  "cursor": "pointer",
                  "marginRight": 24,
                  "textDecoration": "none",
                }
              }
            >
              Sign In
            </span>
          </h2>
          <button
            aria-label="Search"
            className="MuiButtonBase-root-197 MuiIconButton-root-191"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "marginLeft": -12,
                "marginRight": 20,
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-196"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-180"
                color={undefined}
                focusable="false"
                style={
                  Object {
                    "color": "rgb(51, 51, 51)",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-200"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
